keras_core/layers/layer.py

```diff
@@ -646,7 +643,15 @@ def maybe_convert(x):
                 return backend.convert_to_tensor(x, dtype=self.compute_dtype)
             return x
 
-        if self._convert_input_args:
+        if (
```

Comment:
you can just do `if kwargs or len(args) != 1 or not backend.is_tensor(args[0]) or backend.standardize_dtype(args[0].dtype) != self.compute_dtype: args =...`

Reply:
(also I expect a chain of `or` conds to be faster in the negative case because it will exist upon the first match)

Reply:
Done.